src/main/webapp/scripts/private-room-script.js

```diff
@@ -86,44 +85,93 @@ function onYouTubeIframeAPIReady() {
     });
 }
 
-// The API will call this function when the video player is ready.
-function onPlayerReady(event) {
-    document.getElementById('player-div').style.borderColor = '#FF6D00';
-    loadRoomPlaylist();
-}
-
-function stopVideo() {
-    youtubePlayer.stopVideo();
-}
-
 // Log state changes
 function onStateChange(event) {
     var state = "undefined";
     switch (event.data) {
         case YT.PlayerState.UNSTARTED:
-            state = "unstarted";
+            state = "-1";
             break;
         case YT.PlayerState.ENDED:
-            state = "ended";
+            state = "0";
             break;
         case YT.PlayerState.PLAYING:
-            state = "playing";
+            state = "1";
             break;
         case YT.PlayerState.PAUSED:
-            state = "paused";
+            state = "2";
             break;
         case YT.PlayerState.BUFFERING:
-            state = "buffering";
+            state = "3";
             break;
         case YT.PlayerState.CUED:
-            state = "video cued";
+            state = "5";
             break;
         default:
             state = "unknown (" + event.data + ")";
     }
-    console.log('onStateChange: ' + state);
+    timeStamp = Math.round(youtubePlayer.getCurrentTime());
+    updateCurrentState(state, timeStamp);
+}
+
+// Every three seconds you listen to youtube player for any detection of change
+window.setInterval(function(){
+    listenForStateChange();
+}, 1000);
+
+// This function fetches the state of the private room video and
+// plays/pauses it accordingly
+function listenForStateChange(){
+    $(document).ready(function(){
+        const Url = `/sync-room?roomId=${roomId.toString()}`;
+        $.get(Url,function(data, status){
+            // Change timestamp to match group timestamp
+            if(timeStamp + 2 < data.timestamp || timeStamp - 2 > data.timestamp){
+                youtubePlayer.seekTo(data.timestamp);
+            }
+            // Change state to match group state
+            if(data.currentState === "1"){
```

Comment:
This seems a little bit error prone. You can either use int enum value or string type with enum name value. Wrap int as string is not proper.